### ISSUE TITLE
How to integrate this library?

### ISSUE THREAD

**Issue by hasnainusti** [ibrierley/line_animator#5]
Hi, please guide me how can I integrate this library into my project? I cloned the example project and I was able to run it successfully but I don't know how to integrate it into my project. I'm having issue with adding the line animator in pubspec.yaml file.Please help

**Comment by ibrierley** [comment]
Hi, I think you would have something like...

```
flutter_map_line_editor:
  git:
    url: https://github.com/ibrierley/flutter_map_line_editor.git
```

**Comment by hasnainusti** [comment]
Thanks. I've successfully integrated it but when I import the library its giving me this warning:

The library 'package:line_animator/line_animator.dart' is legacy, and should not be imported into a null safe library.
Try migrating the imported library.

**Comment by ibrierley** [comment]
Strange, it should be updated to nullsafety. Are you using the master branch ?

**Comment by hasnainusti** [comment]
No. I'm on 2.5.3. Also getting this error on console .

Error: Cannot run with sound null safety, because the following dependencies
don't support null safety:

 - package:line_animator

For solutions, see https://dart.dev/go/unsound-null-safety

**Comment by ibrierley** [comment]
Strange, it should be updated to null safety, and I've tested it with flutter sdk 2.5.2 and 2.8.0. 

What happens if you run     flutter pub outdated --mode=null-safety ?

(I get direct dependencies: all support null safety. All dependencies opt in to null-safety.)

What does    flutter pub deps show out of interest ?

**Comment by hasnainusti** [comment]
This is the output: 

Showing dependencies that are currently not opted in to null-safety.
[✗] indicates versions without null safety support.
[✓] indicates versions opting in to null safety.

Package Name   Current         Upgradable      Resolvable      Latest          

direct dependencies:
line_animator  ✗1.0.0+1 (git)  ✗1.0.0+1 (git)  ✗1.0.0+1 (git)  ✗1.0.0+1 (git)  

dev_dependencies: all support null safety.
You are already using the newest resolvable versions listed in the 'Resolvable' column.
Newer versions, listed in 'Latest', may not be mutually compatible.

**Comment by ibrierley** [comment]
Doh ignore me, I didn't even read this as the right github repo, I was looking at line_editor which I do, not line_animator. My stupid mistake. Thanks for this. I'll take a look later, probably won't be such a quick fix, but I will take a look when I get time within the next day or so hopefully.

**Comment by ibrierley** [comment]
I've done some updates, so you could try it now.

**Comment by hasnainusti** [comment]
That was quick. Well still getting errors. I think you updated the code but didn't upgrade the environment sdk to the null safety one in pubspec.yaml file.

**Comment by ibrierley** [comment]
Ah yes, well spotted, see if that's better.

**Comment by hasnainusti** [comment]
Thanks. It's working. Can you please also update the example with null safety?

**Comment by ibrierley** [comment]
Is it ok now ?

**Comment by hasnainusti** [comment]
Yes but I'm getting null error on LatLng markerPoint  and List<LatLng> points.

**Comment by ibrierley** [comment]
Sure you have updated it correctly, as line 36 initialises it, so not sure it can be null (and I don't get an error, so seems odd), but maybe I'm looking at something different ?

https://github.com/ibrierley/line_animator/blob/master/example/lib/flutter_map_line_animator.dart

Are you running example/lib/flutter_map_line_animator.dart ?

**Comment by hasnainusti** [comment]
It's working now. I was using the old code. Thanks man

**Comment by ibrierley** [comment]
Super, thanks for the update and the patience.
